### PR TITLE
Smoke test bail apply

### PR DESCRIPTION
**Pull request by squat**
Ensure smoke tests exit if `terraform apply` fails.

Addresses #819

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -62,6 +62,8 @@ pipeline {
             unstash 'installer'
             unstash 'sanity'
             sh '''
+            # Exit if any command fails
+            set -e
```
Comment (ggreer) [inline Jenkinsfile]: Just curious, but do we know what shell `sh` runs? Some of these scripts (like on line 36) have shebangs for `/bin/bash`. They might be using a different shell than steps without shebangs.

Comment (squat) [inline Jenkinsfile]: pretty sure it's straight `sh`

**Comment by ggreer** [comment]
Looks like the regression was introduced in #646.

**Comment by squat** [comment]
mhm.. the issue is that by doing `| filter` we mask the error exit code of the `make apply` so we need `pipefail`

File: Jenkinsfile [hunk 2]
```diff
@@ -61,7 +61,8 @@ pipeline {
                              ]) {
             unstash 'installer'
             unstash 'sanity'
-            sh '''
+            sh """#!/bin/bash -ex -o pipefail
+
```
Comment (ggreer) [inline Jenkinsfile]: You need single quotes here, not double quotes.

Comment (squat) [inline Jenkinsfile]: grr we need to be more consistent across this file

**Comment by ggreer** [comment]
Fails because filter is broken. We should have a separate PR that reverts the filter stuff.

**Comment by kans** [comment]
@squat 
```+ alias filter=/home/ubuntu/workspace/r_tectonic-installer_PR-823-77SS223TD2KXKHELHY34AGVWHYRRI5CZ7N4XGKR33Y5UJ6PQ36FQ/installer/scripts/filter.sh

+ make plan

+ filter

/home/ubuntu/workspace/r_tectonic-installer_PR-823-77SS223TD2KXKHELHY34AGVWHYRRI5CZ7N4XGKR33Y5UJ6PQ36FQ@tmp/durable-93663a4c/script.sh: line 22: filter: command not found
```

At least it didn't run the smoke tests :P

**Comment by squat** [comment]
haha true, at least it stops now! It is failing on filter because we need to explicitly allow alias expansion. Should be good now I think

**Comment by squat** [comment]
ooofta

**Pull request merged**